### PR TITLE
feat(composer): support sync matterport tag as entities

### PR DESCRIPTION
**Pull request by sheilaXu**
## Overview
support sync matterport tag as entities

### Scene Composer
For `scene-composer` package changes specifically, you can preview the component in the published storybook artifact. To do this, wait for the `Publish Storybook` action to complete below.

- Click on the workflow details
- Select the Summary item on the left
- Download the zip file

To run the storybook build locally, you need a local static web server:

```
npm install -g httpserver
cd <Extracted Zip Directory>
httpserver
```

Then open the website http://localhost:8080 to run the doc site.

## Legal
This project is available under the [Apache 2.0 License](http://www.apache.org/licenses/LICENSE-2.0.html).

### CODE REVIEW

File: packages/scene-composer/src/common/entityModelConstants.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { KnownComponentType } from '../interfaces';
+
+// Scene Nodes
+const SCENE_COMPONENT_TYPE_ID_PREFIX = 'com.example.3d';
```
Comment (divya-sea) [inline packages/scene-composer/src/common/entityModelConstants.ts:4]: Maybe we need to come-up with property variable names.

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import { CreateEntityCommandInput } from '@aws-sdk/client-iottwinmaker';
+
+import { SCENE_ROOT_ENTITY_ID } from '../../common/entityModelConstants';
+import { getGlobalSettings } from '../../common/GlobalSettings';
+import { generateUUID } from '../mathUtils';
+
+export const createSceneEntityId = (sceneName: string) => {
```
Comment (divya-sea) [inline packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:7]: Type is missing? That can add onto more warning messages?

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  return `SCENE_${sceneName}_${generateUUID()}`;
+};
+
+export const createSceneRootEntity = () => {
```
Comment (divya-sea) [inline packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:11]: Same?

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  if (!getGlobalSettings().twinMakerSceneMetadataModule) {
+    return;
+  }
+
+  const sceneName = getGlobalSettings().twinMakerSceneMetadataModule!.getSceneId();
+  const sceneEntityId = createSceneEntityId(sceneName);
+  const input: CreateEntityCommandInput = {
+    workspaceId: undefined,
+    entityId: sceneEntityId,
```
Comment (divya-sea) [inline packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:20]: Why name and entityId both takes sceneEntityId?

File: packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+import { createDataBindingMap } from './dataBindingUtils';
+
+export const createTagEntityComponent = (tag: IAnchorComponent): ComponentRequest => {
+  const comp: ComponentRequest = {
+    componentTypeId: componentTypeToId[KnownComponentType.Tag],
+    properties: {},
+  };
+  if (tag.icon) {
+    comp.properties!['icon'] = {
```
Comment (divya-sea) [inline packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts:14]: I guess these exclamation marks can be removed?

File: packages/scene-composer/src/utils/entityModelUtils/sceneLayerUtils.ts [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+              stringValue: type,
+            },
+          },
+        },
+      },
+    },
+  };
+
+  if (type === LayerType.Relationship) {
+    input.components![DEFAULT_LAYER_COMPONENT_NAME].properties![layerPropertyNames.relationshipName] = {
```
Comment (divya-sea) [inline packages/scene-composer/src/utils/entityModelUtils/sceneLayerUtils.ts:41]: What's the reason for using exclamation mark?

Comment (sheilaXu) [reply packages/scene-composer/src/utils/entityModelUtils/sceneLayerUtils.ts:41]: the type of `components` in `CreateEntityCommandInput` can be undefined and typescript is not smart enough to know that I've assigned value to it above

File: packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts (hunk 5 above)
Comment (sheilaXu) [reply packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts:14]: the type of `properties ` in `ComponentRequest` can be undefined and typescript is not smart enough to know that I've assigned value to it above

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts (hunk 4 above)
Comment (sheilaXu) [reply packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:20]: entityName needs to be unique as well

File: packages/scene-composer/src/common/entityModelConstants.ts (hunk 1 above)
Comment (sheilaXu) [reply packages/scene-composer/src/common/entityModelConstants.ts:4]: property name constants? I'll add in each component's file in my next PR

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts (hunk 2 above)
Comment (sheilaXu) [reply packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:7]: right, I'll add in my next PR if no other blocking comment

File: packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts (hunk 3 above)
Comment (sheilaXu) [reply packages/scene-composer/src/utils/entityModelUtils/sceneUtils.ts:11]: I'll add in my next PR if no other blocking comment

File: packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts (hunk 5 above)
Comment (divya-sea) [reply packages/scene-composer/src/utils/entityModelUtils/tagComponent.ts:14]: Maybe we can do this check (comp.properties && comp.properties[icon])?

**Review by divya-sea — APPROVED** [review]
some of the comments can be addressed in next PR.

**Pull request merged**